src/MobileUI/ViewModels/ProfileViewModels/ProfileViewModelBase.cs

```diff
@@ -5,31 +5,42 @@
 using SSW.Rewards.PopupPages;
 using System.Collections.ObjectModel;
 using System.Windows.Input;
+using CommunityToolkit.Mvvm.ComponentModel;
 
 namespace SSW.Rewards.Mobile.ViewModels;
 
-public abstract class ProfileViewModelBase : BaseViewModel, IRecipient<AchievementTappedMessage>
+public partial class ProfileViewModelBase : BaseViewModel, IRecipient<AchievementTappedMessage>
 {
     protected readonly IRewardService _rewardsService;
     protected IUserService _userService;
     protected readonly ISnackbarService _snackbarService;
 
-    public string ProfilePic { get; set; }
-    public string Name { get; set; }
-    public int Points { get; set; }
-    public int Balance { get; set; }
+    [ObservableProperty]
+    private string _profilePic;
+    
+    [ObservableProperty]
+    private string _name;
+    
+    [ObservableProperty]
+    private int _points;
+    
+    [ObservableProperty]
+    private int _balance;
 
     public bool ShowBalance { get; set; } = true;
 
-    public double Progress { get; set; } = 0;
+    [ObservableProperty]
+    private double _progress;
 
     protected int userId { get; set; }
 
-    public bool IsLoading { get; set; }
+    [ObservableProperty]
+    private bool _isLoading;
 
     public bool ShowCamera => _isMe && !IsLoading;
 
-    public ObservableCollection<ProfileCarouselViewModel> ProfileSections { get; set; } = new ObservableCollection<ProfileCarouselViewModel>();
+    [ObservableProperty]
+    private ObservableCollection<ProfileCarouselViewModel> _profileSections = new();
```

Comment:
Not necessary for an ObservableCollection

Reply:
Done - reverted.